docs/debug.md

```diff
@@ -0,0 +1,82 @@
+<!--
+---
+linkTitle: "Debug"
+weight: 11
+---
+-->
+# Debug
+
+- [Overview](#overview)
+- [Debugging TaskRuns](#debugging-taskruns)
+  - [Adding Breakpoints](#adding-breakpoints)
+    - [Breakpoint on Failure](#breakpoint-on-failure)
+      - [Failure of a Step](#failure-of-a-step)
+      - [Halting a Step on failure](#halting-a-step-on-failure)
+      - [Exiting breakpoint](#exiting-breakpoint)
+- [Debug Environment](#debug-environment)
+  - [Mounts](#mounts)
+  - [Debug Scripts](#debug-scripts)
+
+
+## Overview
```

Comment:
Not a blocker but it might be worth mentioning here that debug is behind a feature flag.